alchql/fields.py

```diff
@@ -71,27 +71,36 @@ async def resolve_connection(
         query = await cls.get_query(model, info, **args)
         session = info.context.session
         if resolved is None:
-            only_q = query.with_only_columns(
-                *sa.inspect(model).primary_key,
-            ).order_by(None)
-            if not QueryHelper.get_filters(info) and QueryHelper.has_last_arg(info):
+            has_last_arg = QueryHelper.has_last_arg(info)
+            if not QueryHelper.get_filters(info) and has_last_arg:
                 raise TypeError('Cannot set "last" without filters applied')
 
-            # get max count
+            only_q = only_q = query.with_only_columns(
```

Comment:
only_q = only_q =

Reply:
+move it to `if has_last_arg:`

Reply:
I can't move it under if because I could be used later in case connection has total_count attribute